core/src/main/java/zipkin2/reporter/internal/AsyncReporter.java

```diff
@@ -42,17 +46,18 @@
  *
  * <p>Spans are bundled into messages based on size in bytes or a timeout, whichever happens first.
  *
- * <p>The thread that sends flushes spans to the {@linkplain Sender} does so in a synchronous loop.
- * This means that even asynchronous transports will wait for an ack before sending a next message.
- * We do this so that a surge of spans doesn't overrun memory or bandwidth via hundreds or
- * thousands of in-flight messages. The downside of this is that reporting is limited in speed to
- * what a single thread can clear. When a thread cannot clear the backlog, new spans are dropped.
+ * <p>The thread that sends flushes spans to the {@linkplain BytesMessageSender} does so in a
+ * synchronous loop. This means that even asynchronous transports will wait for an ack before
+ * sending a next message. We do this so that a surge of spans doesn't overrun memory or bandwidth
+ * via hundreds or thousands of in-flight messages. The downside of this is that reporting is
+ * limited in speed to what a single thread can clear. When a thread cannot clear the backlog, new
+ * spans are dropped.
  *
  * @param <S> type of the span, usually {@code zipkin2.Span}
  * @since 3.0
  */
-public abstract class AsyncReporter<S> extends Component implements Reporter<S>, Flushable {
-  public static Builder newBuilder(Sender sender) {
+public abstract class AsyncReporter<S> extends Component implements Reporter<S>, Closeable, Flushable {
+  public static Builder newBuilder(BytesMessageSender sender) {
```

Comment:
This is compile compatible but not sure if linkage compatible, it might be but forget the java details on this. Could be worth a double check

Reply:
I'll put in a test to instantiate a deprecated sender. Good point.

Reply:
PTAL

Reply:
you were right and my tests didn't catch this. invoker could have but anyway https://github.com/openzipkin/zipkin-reporter-java/pull/246